scripts/quality_test/quality_test.py

```diff
@@ -508,37 +271,42 @@ def _extract_missing_pii_annotations(filename):
     return (uncovered_models, pii_check_passed, full_log)
 
 
-@task
-@needs('pavelib.prereqs.install_python_prereqs')
-@cmdopts([
-    ("report-dir=", "r", "Directory in which to put PII reports"),
-])
-@timed
-def run_pii_check(options):
+def run_pii_check():
     """
     Guarantee that all Django models are PII-annotated.
     """
+    REPO_ROOT = repo_root()
+    REPORT_DIR = REPO_ROOT / 'reports'
     pii_report_name = 'pii'
-    default_report_dir = (Env.REPORT_DIR / pii_report_name)
-    report_dir = getattr(options, 'report_dir', default_report_dir)
+    default_report_dir = (REPORT_DIR / pii_report_name)
+    report_dir = default_report_dir
     output_file = os.path.join(report_dir, 'pii_check_{}.report')
     env_report = []
     pii_check_passed = True
     for env_name, env_settings_file in (("CMS", "cms.envs.test"), ("LMS", "lms.envs.test")):
         try:
-            print()
             print(f"Running {env_name} PII Annotation check and report")
             print("-" * 45)
             run_output_file = str(output_file).format(env_name.lower())
-            sh(
-                "mkdir -p {} && "  # lint-amnesty, pylint: disable=duplicate-string-formatting-argument
-                "export DJANGO_SETTINGS_MODULE={}; "
-                "code_annotations django_find_annotations "
-                "--config_file .pii_annotations.yml --report_path {} --app_name {} "
-                "--lint --report --coverage | tee {}".format(
-                    report_dir, env_settings_file, report_dir, env_name.lower(), run_output_file
-                )
+            os.makedirs(report_dir, exist_ok=True)
+            command = (
+                f"export DJANGO_SETTINGS_MODULE={env_settings_file};"
+                f"code_annotations django_find_annotations"
+                f"--config_file .pii_annotations.yml --report_path {report_dir} --app_name {env_name.lower()}"
+                f"--lint --report --coverage | tee {run_output_file}"
+            )
+            result = subprocess.run(
+                command,
+                shell=True,
```

Comment:
Passing a string to `subprocess.run` instead of a list is more error-prone.

In this particular case, if you actually look at the value of `result`, we have:
```
CompletedProcess(args='export DJANGO_SETTINGS_MODULE=cms.envs.test;code_annotations django_find_annotations--config_file .pii_annotations.yml --report_path /openedx/edx-platform/scripts/reports/pii --app_name cms--lint --report --coverage | tee /openedx/edx-platform/scripts/reports/pii/pii_check_cms.report', returncode=0, stdout='', stderr="Usage: code_annotations [OPTIONS] COMMAND [ARGS]...\nTry 'code_annotations --help' for help.\n\nError: No such command 'django_find_annotations--config_file'.\n")
```

The command is failing with `No such command 'django_find_annotations--config_file`, because there are no spaces between your lines. But, the command is not reporting a failure, because when you pipe a failed command into another command, it succeeds! (this is a longstanding pitfall of shell scripting)

To avoid this, I recommend getting rid of `shell=True`, and turn `command` into a list of arguments. You cannot pass in a semicolon or pipe as an argument, though. Instead, you can use `env={**os.environ, "DJANGO_SETTINGS_MODULE": ...}` to set the environment variable, and you can manually open `{run_output_file}` and pass the file handle in as the argument to `stdout`. Please change this across all the quality checks.